### ISSUE TITLE
idk if this a bug or not

### ISSUE THREAD

**Issue by Manglefan** [MrJake222/AUNIS#303]
in the latest beta when i dialed the gate the power drains the longer its open but in the latest alpha when i dial the gate the power doesnt drain or it doesnt lose power till the gate is shut down

**Comment by xnerhu** [comment]
It seems that [this commit](https://github.com/sentialx/AUNIS/commit/7029efdcb88b49583520a0b49e76553ee8996fa6) from @slava110 has broken power draw. After some investigation I found out that `extractEnergy` in `StargateAbstractEnergyStorage` doesn't work because `energy` property in this class is always 0.

@MrJake222

**Comment by xnerhu** [comment]
Also I think it has something with amount of naquadah in DHD. If it's around 0-700 then power draw is 0.

**Comment by Manglefan** [comment]
ive tried with a full DHD and it doesnt work what do i set the StargateAbstractEnergyStorage to

**Comment by MrJake222** [comment]
@slava110

**Comment by slava110** [comment]
> After some investigation I found out that `extractEnergy` in `StargateAbstractEnergyStorage` doesn't work because `energy` property in this class is always 0.

StargateAbstractEnergyStorage wasn't modified in this commit
And energy shouldn't be always 0 if deserialization and serialization works properly
Checked code and it might be the problem
https://github.com/sentialx/AUNIS/blob/7029efdcb88b49583520a0b49e76553ee8996fa6/src/main/java/mrjake/aunis/stargate/power/StargateItemEnergyStorage.java#L26

**Comment by MrJake222** [comment]
Why y'all posting links to the sentialx repo?

**Comment by slava110** [comment]
> Why y'all posting links to the sentialx repo?

He sent link to `AbstractEnergyStorage` class so it's easier for me to click on this link to find the problem

**Comment by xnerhu** [comment]
> > After some investigation I found out that `extractEnergy` in `StargateAbstractEnergyStorage` doesn't work because `energy` property in this class is always 0.
> 
> StargateAbstractEnergyStorage wasn't modified in this commit
> And energy shouldn't be always 0 if deserialization and serialization works properly
> Checked code and it might be the problem
> https://github.com/sentialx/AUNIS/blob/7029efdcb88b49583520a0b49e76553ee8996fa6/src/main/java/mrjake/aunis/stargate/power/StargateItemEnergyStorage.java#L26

I tried a hardcoded value and it didin't work either.

**Comment by xnerhu** [comment]
> Why y'all posting links to the sentialx repo?

Yeah, sorry xd. I'm working with Sentialx to fix and add some stuff.

**Comment by Manglefan** [comment]
So this'll be fixed and how do I add it to my version
